src/nvtxw.rs

```diff
@@ -0,0 +1,394 @@
+        }
+    }
+
+    fn check_info(&mut self) -> Option<DataSourceInfo> {
+        // We requested this once, so we know we'll get zero or one result
+        self.data_source.get_infos().pop()
+    }
+
+    fn write_matched_tile(
+        interface: nvtxw::InterfaceHandle,
+        streams: &BTreeMap<EntryID, nvtxw::StreamHandle>,
+        zero_time: i64,
+        tile: &SlotTile,
+        meta_tile: &SlotMetaTile,
+    ) {
+        assert!(tile.data.items.len() == meta_tile.data.items.len());
+
+        for (row, meta_row) in zip(&tile.data.items, &meta_tile.data.items) {
+            assert!(row.len() == meta_row.len());
+
+            for (item, meta_item) in zip(row, meta_row) {
+                let time_start = item.interval.start;
+                let time_stop = item.interval.stop;
+                let color = item.color;
+                // let time_start = meta_item.original_interval.start;
+                // let time_stop = meta_item.original_interval.stop;
+                let title = meta_item.title.clone();
+
+                let c_name = CString::new(title).expect("CString::new failed");
+                let events = [legion_nvtxw_event {
+                    time_start: (time_start.0 as u64)
+                        .checked_add(zero_time.try_into().unwrap())
+                        .expect("time_start overflowed"),
+                    time_stop: (time_stop.0 as u64)
+                        .checked_add(zero_time.try_into().unwrap())
+                        .expect("time_stop overflowed"),
+                    name: c_name.as_ptr(),
+                    color: ((color.r() as u32) << 16)
+                        | ((color.g() as u32) << 8)
+                        | (color.b() as u32)
+                        | (0xFF << 24),
+                }];
+
+                let stream = streams[&tile.entry_id];
+
+                let payloads = [
+                    nvtxw::PayloadData {
+                        schemaId: LEGION_NVTXW_PAYLOAD_NAME_SCHEMA_ID,
+                        size: usize::MAX,
+                        payload: c_name.as_ptr() as *const c_void,
+                    },
+                    nvtxw::PayloadData {
+                        schemaId: LEGION_NVTXW_PAYLOAD_SCHEMA_ID,
+                        size: size_of::<legion_nvtxw_event>(),
+                        payload: events.as_ptr() as *const c_void,
+                    },
+                ];
+
+                nvtxw::event_write(interface, stream, &payloads).expect("Failed to write event");
+            }
+        }
+    }
+
+    fn process_events(
+        data_source: &mut CountingDeferredDataSource<T>,
+        interface: nvtxw::InterfaceHandle,
+        streams: &BTreeMap<EntryID, nvtxw::StreamHandle>,
+        zero_time: i64,
+        unmatched_tiles: &mut UnmatchedTileHold,
+        num_requests: u64,
+    ) {
+        while data_source.outstanding_requests() > num_requests {
+            let _summary_tiles = data_source.get_summary_tiles();
```

Comment:
I'm not sure we need this since we don't actually use summary tiles? I think it's ok to delete this outright.

Reply:
I turned this back into a comment and added another comment before it.